src/components/Header/utils.ts

```diff
@@ -0,0 +1,255 @@
+import { Navigation, Page, Post } from '@/payload-types'
+import invariant from 'tiny-invariant'
+
+export type NavLink =
+  | {
+      type: 'internal'
+      label: string
+      url: string
+      newTab?: undefined
+    }
+  | {
+      type: 'external'
+      label: string
+      url: string
+      newTab: boolean
+    }
+
+export type NavItem = {
+  id: string
+  link?: NavLink
+  items?: NavItem[]
+}
+
+export type TopLevelNavItem =
+  | {
+      link: NavLink
+      label?: string
+      items?: NavItem[]
+    }
+  | {
+      label: string
+      link?: NavLink
+      items?: NavItem[]
+    }
+
+/**
+ * Convenience function to validate and convert a payload topLevelNavTab to a frontend TopLevelNavItem.
+ *
+ * @returns TopLevelNavItem in an array if the topLevelNavTab is valid or an empty array if invalid.
+ */
+function topLevelNavItem(
+  tab: Navigation['weather' | 'education' | 'accidents' | 'about' | 'support'],
+  label: string,
+): TopLevelNavItem[] {
+  if (!tab) {
+    return []
+  }
+
+  const result: TopLevelNavItem = {
+    label,
+  }
+
+  if (tab.items && tab.items.length > 0) {
+    result.items = tab.items
+      .map((item) => {
+        if (!item) return null
+
+        invariant(item.id, `Tab ${label} has an item without an id.`)
+
+        const navItem: NavItem = {
+          id: item.id,
+        }
+
+        if (item.link) {
+          const convertedLink = convertToNavLink(item.link, [label.toLowerCase()])
+          if (convertedLink) {
+            navItem.link = convertedLink
+          }
+        }
+
+        if (item.items && item.items.length > 0) {
+          navItem.items = item.items
+            .map((nestedItem) => {
+              if (!nestedItem) return null
+
+              invariant(nestedItem.id, `Tab ${label} > item ${item.id} has an item without an id.`)
```

Comment:
```suggestion
              invariant(nestedItem.id, `Tab ${label}[${item.label}][${index}] has no id.`)
```

Should be sufficient info to quickly track down exactly the misbehaving data